core/src/main/java/hudson/os/PosixAPI.java

```diff
@@ -22,6 +22,7 @@
     /**
      * Load the JNR implementation of the POSIX APIs for the current platform.
      * Runtime exceptions will be of type {@link PosixException}.
+     * {@link IllegalStateException} will be thrown for methods not implemented on this platform.
```

Comment:
why not @throws  (for both types of exceptions) - and put them in the method signature?